### ISSUE TITLE
F821 triggers when referencing a deleted (in an if) variable in an else/elif

### ISSUE THREAD

**Issue by jack1142** [charliermarsh/ruff#149]
Example reproduction code:
```py
def func(seq, arg):
    if seq:
        del arg
    elif arg:  # false-positive F821
        print(arg)  # false-positive F821
    else:
        print(arg)  # false-positive F821

    print(arg)  # correctly reported F821
```

---

*Story time*

While running into this false-positive, I actually found a different issue so it's net-positive :)
```py
def func(seq, show_this_once):
    if seq:
        for elem in seq:
            print(elem, show_this_once)
            del show_this_once
    elif show_this_once:
        print(show_this_once)
```
The `del` in the `for` loop here was problematic because on second iteration the variable no loner exists. Somehow we never tested it with a sequence longer than one so this went undetected. Note that ruff didn't report this directly, I just found it because of the `elif` branch I had :) I suppose there could be a rule for reporting this kind of possibly unbound variable issue but it would probably be prone to false positives so I'm not really feature requesting that :smile:

**Comment by jack1142** [comment]
Would be great if early returns were supported as well:
```py
def func(remove, arg1, arg2, arg3):
    if remove & 1:
        del arg1
        return

    print(arg1)

    if remove & 2:
        del arg2
        raise RuntimeError("!!!")

    print(arg2)

    for _ in range(2):
        if remove & 4:
            del arg3
            continue
        print(arg3)
```
as in, all `print`s should be considered valid.
